### ISSUE TITLE
Support CAcert server-side certificates

### ISSUE THREAD

**Issue by mart-e** [e14n/pump.io#786]
When I try to connect to e14n.com with my account from my instance u.mart-e.be, I have got the following error :

```
Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
    at SecurePair.<anonymous> (tls.js:1350:32)
    at SecurePair.EventEmitter.emit (events.js:92:17)
    at SecurePair.maybeInitFinished (tls.js:963:10)
    at CleartextStream.read [as _read] (tls.js:463:15)
    at CleartextStream.Readable.read (_stream_readable.js:320:10)
    at EncryptedStream.write [as _write] (tls.js:366:25)
    at doWrite (_stream_writable.js:219:10)
    at writeOrBuffer (_stream_writable.js:209:5)
    at EncryptedStream.Writable.write (_stream_writable.js:180:11)
    at write (_stream_readable.js:573:24)
    at flow (_stream_readable.js:582:7)
```

I used to be able to connect to this instance but not anymore for a few weeks. I have this also with other instances from e14n (microca.st, 1realtime.net, ...)

**Comment by dper** [comment]
Responding to a old comment about why StartSSL is bad.  It is bad because revocation costs money.  You can do half of what you want for free, but then you find yourself trapped.  You might call it a lock-in SSL provider.

One problem is how to handle certificates.  If I set up a server using CACert, which I find easy to use, then other servers might not talk to me.  But there are no visible error messages -- we have to go looking for them.  This is in contrast with a web browser where an alert appears.

In other words, if pump.io is to be restrictive about certificates, then it would be nice to make the denied connections more visible, and perhaps it would be nice to be able to quickly whitelist certificates or certificate authorities.

To make federation dependent on playing in the big leagues where most people purchase commercial certificates is something to avoid, if we can reasonably do so.

**Comment by strugee** [comment]
This feels like it will be a non-issue when Let's Encrypt rolls out, though.

**Comment by larjona** [comment]
Let's Encrypt is public now. Comments?

**Comment by dper** [comment]
With clear documentation, this solves the problem.  Something like (or better than) this might be good...

> If you are looking for a free SSL certificate, please consider Let's Encrypt.  Self-signed certificates and certificates signed by less-trusted providers like CACert appear to work when you're installing, but later your server may have trouble communicating with other pump servers.

In a perfect world we would also like for people to easily see an error message in the case that they're using a broken or unsigned certificate, but that is a separate and far less important issue -- something like "error information should be more visible".

**Comment by strugee** [comment]
> In a perfect world we would also like for people to easily see an error message in the case that they're using a broken or unsigned certificate, but that is a separate and far less important issue -- something like "error information should be more visible".

Filed #1150.

**Comment by larjona** [comment]
Filed #1151 for the documentation improvement part.
I think this issue can be closed. If there are no further comments here, I'll close it in some days.

**Comment by strugee** [comment]
Closing per @larjona's comment above. The answer to this is now "use Let's Encrypt".
